ddl/index.go

```diff
@@ -1144,7 +1126,12 @@ func (w *addIndexWorker) handleBackfillTask(d *ddlCtx, task *reorgIndexTask) *ad
 			return result
 		}
 
-		taskCtx, err := w.backfillIndexInTxn(handleRange)
+		var taskCtx backfillTaskContext
+		if worker, ok := bf.(*addIndexWorker); ok {
+			taskCtx, err = worker.BackfillDataInTxn(handleRange)
+		} else if worker, ok := bf.(*updateColumnWorker); ok {
+			taskCtx, err = worker.BackfillDataInTxn(handleRange)
+		}
```

Comment:
Can we utilize the method dispatch?
```
taskCtx, err = bf.BackfillDataInTxn(handleRange)
```

Reply:
No, we can't. We need to do some checks when adding indexes, and the filled records have different formats.

Reply:
Got it.